src/inc/pedecoder.inl

```diff
@@ -1100,27 +1100,47 @@ inline void PEDecoder::GetNativeILPEKindAndMachine(DWORD* pdwKind, DWORD* pdwMac
         *pdwMachine = pNativeHeader->Machine;
 }
 
-inline CORCOMPILE_DEPENDENCY * PEDecoder::GetNativeDependencies(COUNT_T *pCount) const
+#endif  // FEATURE_PREJIT
+
+inline CORCOMPILE_DEPENDENCY * PEDecoder::GetNativeDependencies(COUNT_T* pCount) const
 {
     CONTRACTL
     {
         INSTANCE_CHECK;
-        PRECONDITION(CheckNativeHeader());
+        PRECONDITION(HasReadyToRunHeader() || CheckNativeHeader());
         NOTHROW;
         GC_NOTRIGGER;
     }
     CONTRACTL_END;
 
-    IMAGE_DATA_DIRECTORY *pDir = &GetNativeHeader()->Dependencies;
+    IMAGE_DATA_DIRECTORY *pDir;
+
+    if (HasReadyToRunHeader())
+    {
+        pDir = GetReadyToRunSection(READYTORUN_SECTION_NATIVE_DEPENDENCIES);
+        
+        // The section won't be present if there was no version bubble spanning assemblies.
+        if (pDir == NULL)
+        {
+            IMAGE_DATA_DIRECTORY dummy = { 0 };
+            pDir = &dummy;
```

Comment:
This is dangling pointer after the `{`. The compiler is free to reuse the stack space taken by `dummy` for compile else.

Reply:
Oh yeah, I was planning to come back to that.

This is just because the `CONTRACT` macro hell won't let me do a `return` from here and I already had enough of fighting dac macros at that point in the day when I was writing this.